hs_core/hydroshare/utils.py

```diff
@@ -270,22 +270,22 @@ def get_fed_zone_files(irods_fnames):
 
 
 # TODO: make the local cache file (and cleanup) part of ResourceFile state?
-def get_file_from_irods(res_file, temp_dir=None):
+def get_file_from_irods(resource, file_path, temp_dir=None):
     """
-    Copy the file (res_file) from iRODS (local or federated zone)
+    Copy the file (given by file_path) from iRODS (local or federated zone)
     over to django (temp directory) which is
-    necessary for manipulating the file (e.g. metadata extraction).
+    necessary for manipulating the file (e.g. metadata extraction, zipping etc.).
```

Comment:
I don't think it is a good idea to transfer files over from iRODS to Django server for zipping purpose since iRODS can zip file in place. We have a zipup function in django irods storage.py class as shown here: https://github.com/hydroshare/hydroshare/blob/develop/django_irods/storage.py#L121 which can be used to let iRODS zip files in place. Moving files from iRODS to Django for zipping is very costly especially for large files.

Reply:
@hyi I too was bit hesitant in moving files out of iRODS for zipping. It seems the `zipup' function needs to be provided a folder to zip. I assume to do the in place zipping a new temp folder needs to be created under the resource collection and copy the aggregation files to that temp folder for zipping. Then the zip file needs to be moved to the appropriate location under resource collection. Finally this new zip file needs to be registered in Django. Do these steps seem right or  is there a better way to do it?

Reply:
@pkdash You may refer to this function create_temp_zip() in https://github.com/hydroshare/hydroshare/blob/d633226430fd5a4f93d6aa137cb95ec7bbb088ad/hs_core/tasks.py#L278 which appears to be able to zip aggregations. @sblack-usu wrote this function I believe.